src/opendr/perception/object_detection_2d/gem/algorithm/datasets/transforms.py

```diff
@@ -0,0 +1,622 @@
+        right_high_factor = random.uniform(right_high_ratio[0]*h, right_high_ratio[1]*h)
+
+        tl = (0, left_high_factor)
+        bl = (0, left_high_factor+left_low_factor)
+
+        tr = (w, right_high_factor)
+        br = (w, right_high_factor+right_low_factor)
+
+        contour = np.array([tl, tr, br, bl], dtype=np.int32)
+
+        mask = np.zeros([h, w, 3], np.uint8)
+        cv2.fillPoly(mask, [contour], (255, 255, 255))
+        inverted_mask = cv2.bitwise_not(mask)
+        # we need to convert this cv2 masks to PIL images
+        # img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
+        # we skip the above convertion because our mask is just black and white
+        mask_pil = Image.fromarray(mask)
+        inverted_mask_pil = Image.fromarray(inverted_mask)
+
+        low_brightness = F.adjust_brightness(img, low_bright_factor)
+        low_brightness_masked = ImageChops.multiply(low_brightness, mask_pil)
+        high_brightness = F.adjust_brightness(img, high_bright_factor)
+        high_brightness_masked = ImageChops.multiply(high_brightness, inverted_mask_pil)
+
+        return ImageChops.add(low_brightness_masked, high_brightness_masked)
+
+    def __call__(self, img, target, seed):
+        random.seed(seed)
+        if random.uniform(0, 1) < self.p:
+            img = self.process(img, self.high_ratio, self.low_ratio, self.left_low_ratio,
+                               self.left_high_ratio, self.right_low_ratio, self.right_high_ratio, seed)
+            return img, target
+        else:
+            return img, target
+
+
+class RedChannel(object):
+
+    def __init__(self, *args, **kwargs):
+        self.red = RedChannel_class(*args, **kwargs)
+
+    def __call__(self, img, target, seed):
+        return self.red(img), target
+
+
+class RedChannel_class(object):
+    def __init__(self, num_output_channels=1):
+        self.num_output_channels = num_output_channels
+
+    def __call__(self, img):
+        red, _, _ = img.split()
+        image = red.convert("RGB")
+        return image
+
+    def __repr__(self):
+        return self.__class__.__name__ + '(num_output_channels={0})'.format(self.num_output_channels)
+
+
+class RandomSelect(object):
+    """
+    Randomly selects between transforms1 and transforms2,
+    with probability p for transforms1 and (1 - p) for transforms2
+    """
+
+    def __init__(self, transforms1, transforms2, p=0.5):
+        self.transforms1 = transforms1
+        self.transforms2 = transforms2
+        self.p = p
+
+    def __call__(self, img, target, seed):
+        random.seed(seed)
+        if random.random() < self.p:
+            return self.transforms1(img, target, seed)
+        return self.transforms2(img, target, seed)
+
+
+class ToTensor(object):
+    def __call__(self, img, target, seed=None):
+        return F.to_tensor(img), target
+
+# def patch_brightness(img: Tensor, i: int, j: int, h: int, w: int, f: float, inplace: bool = False) -> Tensor:
```

Comment:
This file has several lines of commented code. Might be worth double checking what can be removed and what can and should stay.